### PR TITLE
add functionality to bpf link api

### LINKED ISSUE #199: Add bpf link functionality
As part of our effort to reach feature parity with libbpf-1.0, add a libbpf-rs interface for interacting with bpf links:

LIBBPF_API struct bpf_link *bpf_link__open(const char *path);
LIBBPF_API const char *bpf_link__pin_path(const struct bpf_link *link);
LIBBPF_API void bpf_link__disconnect(struct bpf_link *link);
LIBBPF_API int bpf_link__detach(struct bpf_link *link);

(https://github.com/libbpf/libbpf/blob/3591deb9bc6b0848721f831790b78ae0a593b4c5/src/libbpf.h#L378-L387)

### PR DESCRIPTION
**Pull request by kckeiks**
Add wrapper for bpf_link__open, bpf_link__pin_path and bpf_link__detach.

Fixes #199

### CODE REVIEW

File: libbpf-rs/src/link.rs [hunk 1]
```diff
@@ -15,6 +16,17 @@ impl Link {
         Link { ptr }
     }
 
+    /// Create link from BPF FS file.
+    pub fn open<P: AsRef<Path>>(&mut self, path: P) -> Result<Self> {
+        let path_c = util::path_to_cstring(path)?;
+        let path_ptr = path_c.as_ptr();
+        let ptr = unsafe { libbpf_sys::bpf_link__open(path_ptr) };
+        if ptr.is_null() {
```
Comment (chenhengqi) [inline libbpf-rs/src/link.rs:24]: Currently, the libbpf 1.0 mode is not enabled by default. You should use `libbpf_get_error` here.

Comment (kckeiks) [reply libbpf-rs/src/link.rs:24]: I can change this. 

When will we be using libbpf 1.0 btw? I figured that given the milestone we'd be using that version.

Comment (insearchoflosttime) [reply libbpf-rs/src/link.rs:24]: The libbpf library isn't on 1.0 yet (so libbpf-rs can't using 1.0).

This is determined here: https://github.com/libbpf/libbpf/blob/6edaacad4f7c31f08f63205e2b3569b9f946c8cb/src/libbpf_internal.h#L474 
https://github.com/libbpf/libbpf/blob/3591deb9bc6b0848721f831790b78ae0a593b4c5/src/libbpf.c#L154-L155

**Review by insearchoflosttime — COMMENTED** [review]
Other than what @chenhengqi  mentioned, this looks good to go!

**Review by insearchoflosttime — APPROVED** [review]

**Pull request merged**